src/main/16-cloudwatch-alarms.tf

```
@@ -0,0 +1,141 @@
+locals {
+  microservices = ["interop-probing-api", "interop-probing-caller", "interop-probing-eservice-operations",
+    "interop-probing-eservice-registry-reader", "interop-probing-eservice-registry-updater",
+  "interop-probing-response-updater", "interop-probing-scheduler", "interop-probing-telemetry-writer", "interop-probing-statistics-api"]
+
+  lambda_functions = [
+    aws_lambda_function.cognito_authorizer.function_name,
+    aws_lambda_function.external_authorizer.function_name,
+    aws_lambda_function.cognito_messaging.function_name,
+  ]
+  sqs_queues = [
+    module.sqs_registry_queue.queue_name,
+    module.sqs_polling_queue.queue_name,
+    module.sqs_polling_result_queue.queue_name,
+    module.sqs_telemetry_result_queue.queue_name
+  ]
+
+}
+
+resource "aws_cloudwatch_metric_alarm" "sqs_message_age" {
+  for_each            = toset(local.sqs_queues)
+  treat_missing_data  = "notBreaching"
+  alarm_name          = "${var.app_name}-sqs-message-age-${each.value}-${var.env}"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = 10
+  metric_name         = "ApproximateAgeOfOldestMessage"
+  namespace           = "AWS/SQS"
+  period              = 60
+  statistic           = "Average"
+  dimensions = {
+    QueueName = each.value
+  }
+  threshold     = var.cw_alarm_thresholds.sqs_message_age
+  alarm_actions = [aws_sns_topic.cw_alarms.arn]
+}
+
+resource "aws_cloudwatch_metric_alarm" "lambda_errors" {
+  for_each            = toset(local.lambda_functions)
+  treat_missing_data  = "notBreaching"
+  alarm_name          = "${var.app_name}-${each.value}-errors-${var.env}"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = 10
+  metric_name         = "Errors"
+  namespace           = "AWS/Lambda"
+  period              = 60
+  statistic           = "Sum"
+
+  dimensions = {
+    FunctionName = each.value
+  }
+  threshold     = 1
+  alarm_actions = [aws_sns_topic.cw_alarms.arn]
+}
+
+resource "aws_cloudwatch_metric_alarm" "lambda_concurrency_pct" {
+  treat_missing_data  = "notBreaching"
+  alarm_name          = "${var.app_name}-lambda-conc-util-pct-${var.env}"
+  comparison_operator = "GreaterThanUpperThreshold"
+  evaluation_periods  = 1
+  threshold_metric_id = "sq"
+  alarm_actions       = [aws_sns_topic.cw_alarms.arn]
+
+  metric_query {
+    id          = "sq"
+    expression  = "SERVICE_QUOTA(conc_util)"
+    return_data = true
+    label       = "LambdaConcurrencyUtilization"
+  }
+
+  metric_query {
+    id          = "conc_util"
+    return_data = true
```

```suggestion
    return_data = false
```
Only one metric query can return data